check_limits.py

```diff
@@ -1,18 +1,36 @@
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+def Temperature_check(temperature):
+    if temperature < 0 or temperature > 45:
+        print('Temperature is out of range!')
+        return False
+
+def State_of_charge_check(soc):
+    if soc < 20 or soc > 80:
```

Comment:
usage of hard coded values will lead to issues during maintenance. if it gets used in many places and then a change in one place needs to be consistent with all others. There is a scope for improvement here.